analyses/upstream-analysis/util/function-run-QC.R

```diff
@@ -0,0 +1,66 @@
+###########################################################################
+#' Function to run miQC
+#' @param sce
+#'
+#' @return
+#' @export
+#'
+#' @examples
+run_miQC <- function(sce) {
+  if (any(sce$percent.mito == 0)) {
+    print("miQC did not work, use two steps filtering as defined by `params` for `min_genes` and `mtDNA_pct_default`, and the `FindOutlierThershold` function!")
+    } else {
+      print("miQC worked!")
+      model <- mixtureModel(sce)
+      print(is.nan(model@logLik))    # Apply is.nan function
+      parameters(model)
+      head(posterior(model))
+      print(plotModel(sce, model)) # Plot
+      print(plotFiltering(sce, model))
+      
+      # Convert sce back to seurat object 
+      filtered_sce <- as.Seurat(filterCells(sce, model), 
+                                counts = "counts",
+                                data = "logcounts")
+  }
+  return(filtered_sce)
+}
+
+
+###########################################################################
+#' Function to run QC with default parameters
+#' @param sce
+#'
+#' @return
+#' @export
+#'
+#' @examples
+run_QC_default <- function(sce) {
```

Comment:
Perhaps I am misunderstanding how this function will be called, but based on the code it seems like parameters for use_only_step1 and seurat_obj should be passed to the function?